### PR TITLE
Core/Gossip: Add/Remove spell for disable/enable experience in group

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Add spell for enable/disabling experience gain

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -13909,13 +13909,19 @@ void Player::OnGossipSelect(WorldObject* source, uint32 gossipListId, uint32 men
             SendRespecWipeConfirm(guid, sWorld->getBoolConfig(CONFIG_NO_RESET_TALENT_COST) ? 0 : GetNextResetTalentsCost(), SPEC_RESET_PET_TALENTS);
             break;
         case GossipOptionNpc::DisableXPGain:
+        {
             PlayerTalkClass->SendCloseGossip();
+            CastSpell(source->ToPlayer(), 206662, true); // Experience Eliminated
```
Comment (Aokromes) [inline src/server/game/Entities/Player/Player.cpp:13914]: no magic numbers :)

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -13909,13 +13909,19 @@ void Player::OnGossipSelect(WorldObject* source, uint32 gossipListId, uint32 men
...
             SetPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN);
             break;
+        }
         case GossipOptionNpc::EnableXPGain:
+        {
             PlayerTalkClass->SendCloseGossip();
+            RemoveAurasDueToSpell(206662); // Experience Eliminated
```
Comment (Aokromes) [inline src/server/game/Entities/Player/Player.cpp:13921]: same

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (Shauren) [reply src/server/game/Entities/Player/Player.cpp:13914]: Source is NOT player, this will always be nullptr

**Comment by aquadeus** [comment]
fixed feedback

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -13667,11 +13672,11 @@ void Player::PrepareGossipMenu(WorldObject* source, uint32 menuId, bool showQues
                         canTalk = false;
                     break;
                 case GossipOptionNpc::DisableXPGain:
-                    if (HasPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN))
+                    if (HasPlayerFlag(PLAYER_FLAGS_NO_XP_GAIN) || GetLevel() == DEFAULT_MAX_LEVEL)
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp:13675]: Should compare to MaxLevel from updatefields, not DEFAULT_MAX_LEVEL

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/game/Entities/Player/Player.cpp (hunk 3 above)
Comment (aquadeus) [reply src/server/game/Entities/Player/Player.cpp:13675]: could i use Player::IsMaxLevel()?

Comment (Shauren) [reply src/server/game/Entities/Player/Player.cpp:13675]: yes

**Comment by aquadeus** [comment]
done

**Pull request merged**